pkg/storage/couchdb/couchdbstore.go

```diff
@@ -67,6 +67,11 @@ func NewProvider(hostURL string, opts ...Option) (*Provider, error) {
 		return nil, fmt.Errorf(failToInstantiateKivikClientErrMsg, err)
 	}
 
+	ready, err := client.Ping(context.TODO())
+	if err != nil || !ready {
+		return nil, fmt.Errorf(failToPingCouchDB, err)
```

Comment:
Double check this logic here: what happens if error is nil, but ready is false? What error message gets returned?

Reply:
Perhaps this check should be split up so you avoid potentially printing a nil error?

Reply:
fixed